packages/colorpickers/src/styled/ColorSwatch/StyledCheckIcon.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { parseToRgb, readableColor } from 'polished';
+import styled, { ThemeProps, DefaultTheme } from 'styled-components';
+import { retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import CheckIcon from '@zendeskgarden/svg-icons/src/12/check-sm-fill.svg';
+import { IRGBColor } from '../../utils/types';
+
+const COMPONENT_ID = 'colorpickers.colorswatch_check';
+
+interface IStyledCheckIcon {
+  color: string;
+}
+
+const colorStyles = (props: IStyledCheckIcon & ThemeProps<DefaultTheme>) => {
+  const { theme, color } = props;
+  const { alpha } = parseToRgb(color) as IRGBColor;
+  let checkColor = readableColor(color, theme.colors.foreground, theme.colors.background);
+
+  if (alpha !== undefined && alpha < 0.4) {
+    checkColor = theme.colors.foreground;
+  }
+
+  return `
+    color: ${checkColor}
+  `;
+};
+
+export const StyledCheckIcon = styled(CheckIcon)`
+  transition: opacity 0.25s ease-in-out;
```

Comment:
I appreciate the added finesse. But perhaps due to the size, the timing feels very laggy on this – as if I have to _will_ the box to be checked. Somehow, the `0.25s` feels better on the smaller [Checkbox](https://garden.zendesk.com/components/checkbox), but could use speeding up here – sometimes only takes the slightest nudge, like `0.2s`. 🤷 might just be me. Thoughts, @steue?